### DISCUSSION TITLE
LaraGonzo - A repackaged version of Laragon

### DISCUSSION THREAD

**Discussion by husnilkhatimi** [leokhoa/laragon#916] [Show and tell]
LaraGonzo v6 is a repackaged version of Laragon v6, with some modifications and a customized setup (the installer).

![laragonzo_setup_components](https://github.com/user-attachments/assets/564e69f7-115f-4c99-96e0-3b3a10b295cd)

I made these modifications for personal use, enhancing some of Laragon’s functionalities and simplifying tasks. I figured, why not share it with the world? Ja?! :wink:

## Features / Modifications

+ Updated Binaries and Tools
+ Added Mailpit
+ Added Localtonet
+ Updated Packages and  Sites List
+ Other details in repo and release page

**Additional information**

The setup will run only on 64-bit architecture and minimum of Windows 8. It can be installed on top of an existing Laragon 6 installation. If the setup detects an existing laragon.exe, it will run in extraction mode only. And if you need other components, just run the setup again.

See latest release 
[https://github.com/husnilkhatimi/laragonzo](https://github.com/husnilkhatimi/laragonzo)

**Comment by Ringoinmel** [comment]
good job.  Thanks Bro

**Comment by Sophist-UK** [comment]
Good job, however...

1. I think that the BASIC section should go and the versions put in with the others of the same product so e.g. all PHP versions are together in a group. And if we are supporting X64 and x32, then there should be major sections - otherwise if only x64 then let's be upfront and not bother saying x64 in the list. So here is what I would like to see (with a wishlist of stuff) e.g.
   * PHP
      * [x] PHP v8.3.14 Threadsafe (for Apache)
      * [ ] PHP v8.3.14 Non-Threadsafe (for nginx)
      * [ ] PHP v7.4.33 Threadsafe (for Apache)
      * [ ] PHP v7.4.33 Non-Threadsafe (for nginx)
      * [ ] PHP v8.2.26 Threadsafe (for Apache)
      * [ ] PHP v8.2.26 Non-Threadsafe (for nginx)
      * [ ] PHP v8.4.1 Threadsafe (for Apache)
      * [ ] PHP v8.4.1 Non-Threadsafe (for nginx)
   * Web Server
      * [x] Apache v2.4.62
      * [ ] Apache ...
      * [ ] nginx ...
      * [ ] node 22.11 (javascript)
      * Node package managers
         * [ ] npm
         * [ ] yarn
         * [ ] bun
      * [ ] dino (typescript)
      * [ ] Python 3.11
    * SQL Database server
       * [x] MySQL v8.0.40
       * [ ] MySQL v9.0.x
       * [ ] MariaDB 11.4.4
       * [ ] Postgres
       * [ ] SQLite
       * [ ] ...
    * SQL Database Admin
       * [x] Heidi (for MySQL, ...)
       * [x] PHPMySQL (for ...)
       * [ ] DBeaver Community Edition (for ...)
       * [ ] PGAdmin (for ...)
       * [ ] adminer
       * ...
    * Other servers
       * [ ] Redis
       * [ ] memcached
       * [ ] rabbitmq
       * [ ] memio
       * [ ] Mailpit
       * [ ] mongo
       * [ ] mongoadmin
       * [ ] socketi
       * [ ] ...
    * [ ] Code editors (if not already installed - do not duplicate)
       * [ ] Notepad++
       * [ ] VSCode (plus starter plugins for selected products)
       * [ ] ...
    * [ ] Frameworks
       * Laravel (php)
          * [ ] Laravel 10
          * [ ] Laravel 11
          * [ ] Laravel 12 beta
          * [ ] Livewire 3
          * [ ] Filament 3
          * [ ] Jetstream
          * [ ] Breeze
          * [ ] DebugBar
          * [ ] ...
       * [ ] Django (python)
       * [ ] Hapi.js (node)
       * [ ] Express.js (node)
       * [ ] Koa.js (node)
       * [ ] Meteor.js (node)
       * [ ] ...

   Ideally there would also be a web link to a documentation site or release notes page) against each product. Default versions would have boxes pre-checked. 

   The Frameworks section is probably specific to each web-site you are developing, and so might need to be in Laragon itself rather than the installer. Perhaps a "New site" command shell wizard could be run which asks questions and prompts for answers about what you want, and then configures it up.

   **Note 1:** This is an subset of an ultimate wish-list for Laragon to be a **web developers one-stop shop**. I wouldn't expect to see all this happen in one go, and I would imagine it would need to be a community effort.
   **Note 2:** Some of these might not have Windows versions and might only be runnable under e.g. Docker, and that is IMO too complex to include.

2. This also means that we DON'T bundle any versions in with the installer but instead keep the LaraGonzo installer small and:
    * Download each product from the appropriate site and install it into the appropriate directory
    * Configure all necessary configuration files to make it all work together.

    Ideally this would be actually be achieved on first run of the Laragon executable so that the same scripts can be used to do the same installs later, but I suspect that is not possible at present. So instead, ideally we would have a Laragon installer that will look at the existing configuration and set the checkboxes for installed products (with greyed out titles for those that are obsolete) so that they can be unchecked and the installer will remove them.

**Reply by husnilkhatimi** [reply]
Thank you for your inputs.

I use this myself and to share to students, some of them have limited internet access will copy the setup using flashdrive, thats why i created the 'basic' that is bundled in the setup. So even without internet they still can use basic laragon features for learning.

For now I coded the setup to only support x64 because most of the components are for x64.

To include x86 (32bit) the setup size would be double (to include different architecture) and some extra coding to install it seperately, OR create a different setup exe for 32bit.

For other php version of NTS, i included it in packages list. So user can install later by going to Menu - Tools - Quick add

And some of the components you listed are in my wishlist too, but LaraGonzo is still using the same Laragon exe, its still limited to what it is now 😊

**Reply by Sophist-UK** [reply]
My suggestion would be to get the installer to look for a specific cache directory with pre-downloaded products in it - then you can bundle whatever products you want with the installer simply by putting the downloads on the stick as well. Indeed, the installer could look in this directory as an alternative way to decide what checkboxes to default to "on".

**Reply by Sophist-UK** [reply]
I am NOT advocating including 32bit versions - almost everyone who is on Windows is using 10 or 11 and these are 64-bit only. I am just pointing out that if the entire project is only 64-bit, then it is pointless saying it for every product individually.

**Reply by Sophist-UK** [reply]
In other words, I am suggesting to you pretty much the same as I have previously suggested to Leo - aim on building something flexible and easily extended for more products rather than something tailored to a specific set of products that needs to be changed substantially every time you want to change e.g. the default product set.

**Comment by Sophist-UK** [comment]
@leokhoa I really encourage you to see what is happening here - your control of Laragon is eroding because people are no longer prepared to wait for you to have time. If you don't want to end up having someone develop an alternative executable and create a competing and entirely separate open source project, then you need to start being more willing to accept some help.

Please don't get me wrong - IME a community forking and set up a competing project is not a good thing because it splits the community. But there are no shortage of examples where this has happened and the new project is successful whilst the old one dies e.g. OpenOffice=>LibreOffice, Mambo=>Joomla etc. etc. etc. 

Edit: As an example, if things continue this way, sooner or later @husnilkhatimi's extensions to your executable will reach the limit of what can be achieved whilst still using your executable, and then someone will write an alternative executable and the project will then no longer be reliant on anything you have created and will in effect be a separate project altogether.

**Reply by husnilkhatimi** [reply]
He starred my fork too. So he already know about this. 

I really hope he can continue Laragon development soon. So many ideas and functions can be implemented.

**Reply by Sophist-UK** [reply]
As far as I know, **all** the community members (including yourself) just want to help and not create difficulties - but we really need to have a better solution than a time-constrained, uncommunicative one-man band. 

Laragon could easily become a comprehensive one-stop-shop for easily creating web a  local development environment - if only Leo would listen to the ideas of the community and let us help him to achieve this.

**Reply by Lexian-droid** [reply]
> As far as I know, **all** the community members (including yourself) just want to help and not create difficulties - but we really need to have a better solution than a time-constrained, uncommunicative one-man band.
> 
> Laragon could easily become a comprehensive one-stop-shop for easily creating web a local development environment - if only Leo would listen to the ideas of the community and let us help him to achieve this.

I think we lost him already, I've been using Laragon for a long while, use to use Mamp, it stopped being useful, gave me issues with Laravel, etc, and now I'm here, however, I seem to have put my eggs into one basket, as now Leo seems to want to take the project into paid territory, which means my projects now remain doorment until I find a solution, I will continue looking, but I hope LaraGonzo can help me last until I find a viable (open source) competitor. Having a couple issues atm though.

**Reply by Sophist-UK** [reply]
Leo is (very very) slowly coming around to listening to the community - but it does feel like it's happening at a snails pace.

**Comment by PhiDelta-t** [comment]
Wow, great

**Comment by DevOpDan** [comment]
What happens if I already have the original Laragon 6? I'm still actively using it fine, only issue I had was needing to update Apache, so having a single installer to just modernise the lot in one go would be nice.

Edit: " It can be installed on top of an existing Laragon 6 installation." - Serves me right for skim reading. -1 brain cell.

**Comment by aaanadie1** [comment]
Hello,
Excellent work, the only thing I miss from Local by Flywheel is be able to clone and blueprint a site, if you can implement that would be perfect. Thanks a lot.

**Reply by husnilkhatimi** [reply]
Can you share some tools that you are using right now for that purposes? I will try them first

**Comment by imaxenio** [comment]
@husnilkhatimi Good job!! and great contribution!! Just a note, the naming or branding of the project doesn't quite sound very good. Maybe something like LaraServ or LaraHost sounds better. And if you want to differentiate this Fork, I can make you a new logo and send it to you in .svg format or whatever you need for branding, I can even help you create the official landing page of the project.
Good job!! If you need help, the Laragon community is here to help you improve.
Thank you

**Reply by husnilkhatimi** [reply]
Hi, thank you for the ideas and offers. LaraGonzo is primarily a drop-in component upgrader for Laragon v6, designed so that my students and friends can use it right away without needing to manually download new components. It also includes some modifications I’ve made. I don’t intend to create a 'new' competing software or establish a separate community.

The Laragon executable itself has limitations when it comes to major changes and upgrades, as most of its functions are hardcoded into a single executable file. In the future, v7 and newer versions will likely be more appealing to users.

**Comment by Lexian-droid** [comment]
Hello, do you by chance know why this is causing the localhost to redirect to a project? if so, how do I undo it? It goes to a project even though I'm not loading a project.

**Reply by husnilkhatimi** [reply]
Hi, can you explain further what do you mean by 'not loading a project'?

**Reply by Lexian-droid** [reply]
> Hi, can you explain further what do you mean by 'not loading a project'?

So, when I download Laragonzo, it will install without issue, but when I actually run it and set my root directory, then I hit the "Web" button, it takes me to the first project in the directory, that's not a normal thing it should do, Laragon should display a "Default" home page when you click the "Web" button, for me, I modified mine, I use a page with a bunch of helpful resources, buttons, etc as well as links directly to the projects, and a link to open VSCode.

**Reply by husnilkhatimi** [reply]
All of the components (apache, nginx, php, etc) are not touch. Its extracted as it is from the source. you can confirm it if need to, by downloading the same version from their respective websites.

So after installation you had change the root of www folder? Maybe its related to that?

Another thing you can test is, since LaraGonzo just a 'drop-in' components upgrade, you can reinstall original Laragon v6 on top of LaraGonzo installation to see if there any different.

**Reply by husnilkhatimi** [reply]
@Lexian-droid hi i just posted a new released. I included a simple script to standardize vhost config. can you give it a try, install on top of your problematic laragon 6. see whether it fix this problem

**Reply by PhiDelta-t** [reply]
> @Lexian-droid hi i just posted a new released. I included a simple script to standardize vhost config. can you give it a try, install on top of your problematic laragon 6. see whether it fix this problem

Thx sir. Semoga diberikan rezeki yang berlimpah, berkah dan halal

**Reply by Lexian-droid** [reply]
> @Lexian-droid hi i just posted a new released. I included a simple script to standardize vhost config. can you give it a try, install on top of your problematic laragon 6. see whether it fix this problem

Perfection! That works! 😄 Thank you so much!

**Comment by dhanyn10singapay** [comment]
@husnilkhatimi please open issue on your repo. i need to report a problem after installing your apps in my pc.

**Reply by husnilkhatimi** [reply]
Please share it here

**Reply by dhanyn10singapay** [reply]
why here? why not on your own repo?

**Reply by husnilkhatimi** [reply]
Please refer [here](https://github.com/leokhoa/laragon/discussions/916#discussioncomment-11702010)

**Reply by Lexian-droid** [reply]
Hey bro can you [respond to my comment?](https://github.com/leokhoa/laragon/discussions/916#discussioncomment-11781491) just sitting here waiting for a response lol, likewise, i also can't open an issue. You obviously modified something that would cause these issues to happen, as the standard version of Laragon doesn't have any of these issues, so if you would know why this is happening, It would be helpful for me to revert it. Issues on that repo, or hell at least even Discussions would make it easier to ask for help.

**Reply by husnilkhatimi** [reply]
@Lexian-droid sorry hi miss the notification. Actually i can't modified anything inside laragon exe, i can only add sripts for additional functions.

I replied your post.

**Reply by Lexian-droid** [reply]
> @Lexian-droid sorry hi miss the notification. Actually i can't modified anything inside laragon exe, i can only add sripts for additional functions.
> 
> I replied your post.

Indeed you are correct in that you can't modify the Laragon EXE, however mayhaps you modified the configuration to some additional software, like Apache? We can continue this discussion in my thread if you'd like.

**Comment by gathlete** [comment]
the latest [LaraGonzo_v6_b250505.exe] file contains a virus according to avast. the previous version I used LaraGonzo_v6_b250111.exe is ok

**Reply by Sophist-UK** [reply]
It is fairly common for installers to trigger false positives - because malware authors use the same installers to package their malware and some AV products are incapable of separating the packaging tool from the payload.

(I should add that it is often perceived by AV firms to be better-safe-than-sorry, leading to more false positives - and indeed some AV firms use their positive virus identification rates as marketing which further encourages false positives.)

Perhaps you could provide more information about the exact virus / exact message that Avast is giving so that we can try to narrow down whether the virus found is likely / unlikely to be a packaging false positive or not.

**Reply by husnilkhatimi** [reply]
hi, is it showing virus detected or PuP (potentially unwanted program), or unknown ? 

As explained by @Sophist-UK , this same installer software (InnoSetup) used by so many people including threat actor to spread malware, repackaged legit software or game with malware etc. InnoSetup is very famous and free setup builder software. Same one used by FitGirl and other repacker.

If unkown or PuP, sometime antivirus will assume as risky program first since every new setup as its own distinct signature.

**Comment by gathlete** [comment]
[image: Screenshot.png]

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Comment by gathlete** [comment]
i hope this helps

<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
Virus-free.www.avast.com
<https://www.avast.com/sig-email?utm_medium=email&utm_source=link&utm_campaign=sig-email&utm_content=webmail>
<#DAB4FAD8-2DD7-40BB-A1B8-4E2AA1F9FDF2>

**Reply by Sophist-UK** [reply]
No.

It says "Virus free" so now I am completely confused...

**Reply by Sophist-UK** [reply]
And you cannot send a screenshot if you reply by email rather than by accessing this Github discussion using your web browser.

**Reply by gathlete** [reply]
yes i sent the screenshot via gmail.  here it is again
<img width="604" height="412" alt="unnamed" src="https://github.com/user-attachments/assets/d21fc52d-f651-4437-92df-af3c91a806af" />

**Comment by gathlete** [comment]
maybe the best scenario is to repackage?

**Comment by gathlete** [comment]
previously i tried to install. and gave me the first result FileRepPup.. a full scan of the file gave me this...
<img width="1010" height="674" alt="Screenshot 2025-11-18 142947" src="https://github.com/user-attachments/assets/deeef425-3736-4b42-bfb7-50d7315a0b0d" />

**Reply by Sophist-UK** [reply]
Actually there is a lot to unpack here:

1. FileRepPUP is short for File Reputation - Potentially Unwanted Programme.
2. A Potentially Unwanted Programme is NOT malware. Here is the [Wikipedia]() definition:

   > A potentially unwanted program (PUP) or potentially unwanted application (PUA) is software that a user may perceive as unwanted or unnecessary. It is used as a subjective tagging criterion by security and parental control products. Such software may use an implementation that can compromise privacy or weaken the computer's security.

   So the absolute worst case here is that it is installing a programme that you may not want, and **NOT** something that actually contains malware - Avast is using scare tactics here to make you believe that it is saving you from malware when it isn't. 

3. The file it is referring to is the Laragonzo installer - which installs lots of files - and the AV programme is not saying whether it is the installer executable that it is flagging or one of the files it is going to install. The trouble with installers, particularly free ones, is that they are often used by actual malware authors to package up their malware - and then AV products (particularly those that are less capable and don't look past the fact that it is an installer to examine the files it is going to install) can give false-positives just because it is the same installer.

So, as always it is your choice as to whether you install Larazonzo or not - but my advice is:

1. Install it and see if the AV flags up any installed files as PUP or malware (which if it is any good it should do) - and then you might get more details on what and why it is flagging it;
2. Bear in mind that many other people (myself included) have installed this exact release without our (different) AV software flagging it;
3. Based on this decide for yourself whether to allow the software on your machine or not.

Bear in mind that there are other alternatives you can try.

**Comment by ronmarasigan** [comment]
This was detected when I downloaded the file.

Trojan:Win32/Alevaul!MSR
[Detected by Microsoft Defender Antivirus](https://www.microsoft.com/en-us/windows/windows-defender?ocid=cx-wdsi-ency)
Aliases: No associated aliases

Summary
Trojan:Win32/Alevaul!MSR is a trojan downloader that compromises Windows devices. Infection typically occurs when the trojan is bundled with other malware or when a user unknowingly downloads it from a compromised website. The primary objective of this threat is to evade detection, to establish unauthorized access and control, or to collect sensitive user information. Its operations are non-destructive by design, relying instead on stealth tactics such as execution delays and network traffic flooding to avoid security software. 

The initial confirmed sample of this threat has a creation date of March 27, 2025, indicating its recent emergence. Alevaul presents a moderate risk level due to its core function: retrieving and executing additional malicious scripts from remote servers. Although its code is not encrypted, it is heavily obfuscated and uses subtle launch techniques to remain on a device for extended periods. This persistence is achieved through task scheduling and automation. The malware does not function as a rootkit, meaning it does not employ advanced privilege escalation to hide itself within protected system areas.

**Reply by husnilkhatimi** [reply]
As discussed in some posts above, this is very common for InnoSetup installer. High ratio compressing setting also contribute to this false positive detection.

If you want to investigate further, you can try to install it in virtual machine, and full scan with antivirus after that to confirm that it is in fact the setup itself is being detected, not the contents. Some protection software might detect tools like ngrok as threat too.

**Reply by Sophist-UK** [reply]
Literally the comment immediately above this post is the one where I explain in detail why these get false positives. It would be helpful if you could read before posting.

**Comment by JeffGepiga** [comment]
I also built similar to laragon,
check this out -> https://jeffgepiga.github.io/DevBoxPro
Its cross-platform, fully tested on windows, not yet on MAC

**Reply by PhiDelta-t** [reply]
This looks solid. Can't wait to try it out!

**Reply by JeffGepiga** [reply]
> This looks solid. Can't wait to try it out!

Yeah!, let me know if you found any issues!,
Im using this personally for my laravel projects

**Reply by DevOpDan** [reply]
Looks nice, but no Linux?

**Reply by DevJeff27** [reply]
> Looks nice, but no Linux?

Its designed to work on unix but not fully tested, pls try to build it on your machine and let met know. My other account got blocked here, pls create an issue to the repo if you found one

**Comment by dontletthesamedogbiteyoutwice** [comment]
Can I run DevBox side by side with Laragon? Or will there be conflicts? Not at the same time but on the same machine.

**Reply by DevJeff27** [reply]
Yes it should not cause conflict
